src/workloads/execution/SecondaryReadsGenny.yml

```diff
@@ -0,0 +1,231 @@
+SchemaVersion: 2018-07-01
+Owner: "@TODO"  # what should go here? us or the owners of secondary_reads?
+Description: >
+ Test performance of secondary reads while applying oplogs
+
+ ### Setup
+
+ This workload runs only in a replica set with EXACTLY 3 NODES.
+ To use it in a replica set with a different number of nodes, adapt the
+ *number_of_nodes parameter (needs to be >1).
+
+ ### Test
+
+ On an empty replica set, the workload performs the following actions:
+ - Setup stage:
+    - Inserted *initial_nb_docs* documents into both the primary and the
+      secondaries. This should be done before any writes and reads.
+ - Actual test stage:
+    - *nWriters* background writers doing infinite writes on the primary which
+      will replicate these writes to secondaries.
+    - 1/16/32 readers doing queries on a sepcified seconary. Each query has
+      *batch_size* numbers of documents and only reads documents inserted in the
+      setup stage in order to have more constant behavior all the time.
+    - Finally, a query with WriteConcern 3 is used to measure the replication lag
+ - Cleanup stage:
+   - Drop the collection to reset the db SortPattern:
+ The test cycles through these stages for 1, 16, and 32 reader threads.
+
+
+Initial_nb_docs: &initial_nb_docs 10000
+DocumentSize: &document_size 459  # Target document size of 500 and Object.bsonsize({_id: ObjectId(), x: 0, str: "x"}) + 1 = 41
+ReadLimit: &read_limit 100
+MaxIdForFinds: &max_id_for_finds 9900
+TestDurationInSecs: &phase_duration 180 seconds
+Db: &db test
+WriterThreads: &writer_threads 16
+NumPhases: &max_phase 13
+NumberOfNodes: &number_of_nodes 3
+
+# This client allows the ReadsOnSecondary actor to directrly connect to the secondary node.
+# All other actors use the default client to connect to the primary node.
+Clients:
+  Secondary:
+    URI: mongodb://md1:27017
+
+ActorTemplates:
+- TemplateName: WritesOnPrimaryTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: WritesOnPrimary}}
+    Type: CrudActor
+    Threads: *writer_threads
+    Database: *db
+    Phases:
+      OnlyActiveInPhases:
+        Active: {^Parameter: {Name: "ActivePhases", Default: [-1]}}
+        NopInPhasesUpTo: *max_phase
+        PhaseConfig:
+          Duration: *phase_duration
+          Collection: Collection0
+          Operations:
+          - OperationName: insertOne
+            OperationCommand:
+              Document:
+                x: {^RandomInt: {min: *initial_nb_docs, max: 1000000000}}
+                s: {^RandomString: {length: *document_size}}
+
+- TemplateName: ReadsOnSecondaryTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: ReadsOnSecondary}}
+    Type: CrudActor
+    Threads: {^Parameter: {Name: "Threads", Default: -1}}
+    ClientName: Secondary
+    Database: *db
+    Phases:
+      OnlyActiveInPhases:
+        Active: {^Parameter: {Name: "ActivePhases", Default: [-1]}}
```

Comment:
Just for my own knowledge, where can I find the list of phases for this workload?

Reply:
From lines 46-111, the workload defines a number of templates. Those templates are then used in the `ActorFromTemplate` block on lines 161-216, where the phase is passed as an argument. So the ReadsOnSecondary actor is running in phases 2, 7, and 12.

As for a general overview of the phases in this test:
```
0: LoadInitialData
1: Quiesce (fsync, etc)
2: WritesOnPrimary (16 Threads) + ReadsOnSecondary (1 Thread)
3: ReplicationLag (measure replication lag via write with w:3)
4: CleanUp (drop collection)
5: LoadInitialData
6: Quiesce
7: WritesOnPrimary (16 Threads) + ReadsOnSecondary (16 Threads)
8: ReplicationLag
9: CleanUp
10: LoadInitialData
11: Quiesce
12: WritesOnPrimary (16 Threads) + ReadsOnSecondary (32 Threads)
13: Replication lag
```

Reply:
This list of phases has value, can we add it as a comment?

**Also, what does a ActivePhases value of [-1] actually do? I hope it throws an exception but I suspect it does nothing.**